### PR TITLE
EDU 334: update List Workflow information

### PR DESCRIPTION
**Pull request by DJSanti**
## What does this PR do?
This PR adds information about the Go SDK's List Workflow functions.

## Notes to reviewers
There are four separate functions in the Go SDK.

### CODE REVIEW

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 1]
```diff
@@ -9,4 +9,42 @@ tags:
...
-Use [`Client.ListWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow).
+There are several functions for listing [Workflow Executions](/concepts/what-is-a-workflow-execution) available on the [Temporal Go SDK](https://pkg.go.dev/go.temporal.io/sdk).
+Each function retrieves Workflow Executions according to a given [Query](/concepts/what-is-a-query) or request filters.
+
+To list all Workflow Executions, use [`ListWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow) on the [Temporal Client](/concepts/what-is-a-temporal-client).
+
+```go
+resp, err := temporalClient.ListWorkflow(context.Background(), sampleListRequest)
+
+// print?
```
Comment (flossypurse) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:20]: do we need these print comments?

Comment (DJSanti) [reply docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:20]: Still trying to figure out if I need to include a print statement for the response or if it just does it of its own accord.

Comment (DJSanti) [reply docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:20]: I wasn't sure, so I'm gonna ask Roey or somebody

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 2]
```diff
@@ -9,4 +9,70 @@ tags:
   - client
 ---
 
-Use [`Client.ListWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow).
+There are several functions for listing [Workflow Executions](/concepts/what-is-a-workflow-execution) available on the [Temporal Go SDK](https://pkg.go.dev/go.temporal.io/sdk).
+Each function retrieves Workflow Executions according to a given [Query](/concepts/what-is-a-query) or request filters.
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:13]:
```suggestion
Several functions for listing [Workflow Executions](/concepts/what-is-a-workflow-execution) are available on the [Temporal Go SDK](https://pkg.go.dev/go.temporal.io/sdk).
Each function retrieves Workflow Executions according to a specified [Query](/concepts/what-is-a-query) or request filters.
```

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 3]
```diff
@@ -9,4 +9,70 @@ tags:
...
+
+fmt.Println("First page of results:")
+for _, exec := range resp.Executions {
+  fmt.Println("Workflow ID %v\n", exec.Execution.WorkflowId)
+  fmt.Println("Next page of results:")
+}
+```
+
+To return a list of archived Workflow Executions, use `ListArchivedWorkflow()` on the Client.
+Make sure that the Temporal Cluster or target [Namespace](/concepts/what-is-a-namespace) is configured for [visibility archival](/concepts/what-is-standard-visibility) before using this function.
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:31]:
```suggestion
Make sure that the Temporal Cluster or target [Namespace](/concepts/what-is-a-namespace) is configured for [Archival](/concepts/what-is-archival) before using this function.
```
Is this change accurate? (I could also make a case for linking to https://temporal-documentation-ao26cn23g.preview.thundergun.io/cluster-deployment-guide#set-up-archival, but that information is only one click away from the definition of Archival.)

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 4]
```diff
@@ -9,4 +9,25 @@ tags:
...
+```go
+resp, err := temporalClient.ListWorkflow(ctx.Background(), req)
+if err != nil {
+  return err
+}
+
+fmt.Println("First page of results:")
+for _, exec := range resp.Executions {
+  fmt.Println("Workflow ID %v\n", exec.Execution.WorkflowId)
+  fmt.Println("Next page of results:")
```
[the remark this answers is not in the document]
Comment (DJSanti) [reply docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:31]: I'm going to take out the paging and just have it as a list.

**Review by djmagee — APPROVED** [review]

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 5]
```diff
@@ -9,4 +9,31 @@ tags:
   - client
 ---
 
-Use [`Client.ListWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow).
+The `ListWorkflow()` function retrieves [Workflow Executions](/concepts/what-is-a-workflow-execution) that match the properties specified in a [List Filter](/concepts/what-is-a-list-filter).
+These properties are defined by [Search Attributes](/concepts/what-is-a-search-attribute), which are used to retrieve Workflow Executions from the [Visibility store](/concepts/what-is-visibility).
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:13]: Should `Visibility store` link to `/cluster-deployment-guide#visibility-store`?

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 6]
```diff
@@ -9,4 +9,31 @@ tags:
...
+
+Use a List Filter to define a `request` to pass into `ListWorkflow()`.
+
+```go
+// ...
+request := "CloseTime = missing"
+```
+
+This `request` value will only return open Workflows.
+To list all Workflow Executions that satisfy this List Filter, use [`ListWorkflow()`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow) on the [Temporal Client](/concepts/what-is-a-temporal-client).
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:23]:
```suggestion
To list all Workflow Executions that satisfy this List Filter, use [ListWorkflow()](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow) on the [Temporal Client](/concepts/what-is-a-temporal-client).
```

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 7]
```diff
@@ -9,4 +9,31 @@ tags:
...
+These properties are defined by [Search Attributes](/concepts/what-is-a-search-attribute), which are used to retrieve Workflow Executions from the [Visibility store](/concepts/what-is-visibility).
+
+Use a List Filter to define a `request` to pass into `ListWorkflow()`.
+
+```go
+// ...
+request := "CloseTime = missing"
+```
+
+This `request` value will only return open Workflows.
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:22]:
```suggestion
This `request` value returns only open Workflows.
```

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 8]
```diff
@@ -9,4 +9,31 @@ tags:
...
+if err != nil {
+  return err
+}
+
+fmt.Println("List of results:")
+for _, exec := range resp.Executions {
+  fmt.Println("Workflow ID %v\n", exec.Execution.WorkflowId)
+}
+
+//...
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:38]:
```suggestion
// ...
```
Or remove the space from the first line. I don't think we've defined a style yet, so let's just aim for consistency.

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md (hunk 5 above)
Comment (flossypurse) [reply docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:13]: I think the specificity is a little bit off...

WFEs are not stored in the visibility store - they are stored in the normal persistence. Search attribute data is stored in the visibility store (what search attributes are part of the system basically)

WFEs are stored in normal persistence.
Search Attribute source of truth data is in the Vis store.
The ListWorkflow function is used to retrieve a list of WFs
The List Filter provided to the ListWorkflow call filters and sorts the list.
What is returned is WFE meta data - 
You can use that to get the full details and history of and individual WFE.

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md (hunk 7 above)
Comment (flossypurse) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:22]: More List Filter syntax examples can be found... and link to that spot?

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md (hunk 6 above)
Comment (flossypurse) [reply docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:23]: This is a bit verbose and repetitive -
You could wipe this whole sentence if the API definition is linked in the first mention of the ListWorkflow API in the first sentence.

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 9]
```diff
@@ -9,4 +9,26 @@ tags:
...
 ---
 
-Use [`Client.ListWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow).
+The [`ListWorkflow()`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow) function retrieves a list of [Workflow Executions](/concepts/what-is-a-workflow-execution) that match the [Search Attributes](/concepts/what-is-a-search-attribute) of a given [List Filter](/concepts/what-is-a-list-filter).
+The metadata returned from the [Visibility store](/concepts/what-is-visibility) can be used to get a Workflow Execution's history and details from the [Persistence store](/concepts/what-is-a-temporal-cluster#persistence).
+
+Use a List Filter to define a `request` to pass into `ListWorkflow()`.
+
+```go
+request := "CloseTime = missing"
```
Comment (cretz) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:18]: `request` has to be a full `ListWorkflowExecutionsRequest` object here, not just a string, so something like:

```suggestion
request := &workflowservice.ListWorkflowExecutionsRequest{ Query: "CloseTime = missing" }
```

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 10]
```diff
@@ -9,4 +9,26 @@ tags:
...
+This `request` value returns only open Workflows.
+For more List Filter examples, see the [examples provided for List Filters in Temporal's Visibility guide.](/concepts/what-is-a-list-filter#list-filter-examples)
+
+```go
+resp, err := temporalClient.ListWorkflow(ctx.Background(), request)
+if err != nil {
+  return err
+}
+
+fmt.Println("List of results:")
```
Comment (cretz) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:30]:
```suggestion
fmt.Println("First page of results:")
```

I liked that you made it clear before that this only the first page of results. If users want more than the first page, they have to do extra work (they have to re-invoke).

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 11]
```diff
@@ -9,4 +9,26 @@ tags:
...
+for _, exec := range resp.Executions {
+  fmt.Println("Workflow ID %v\n", exec.Execution.WorkflowId)
```
Comment (cretz) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:32]:
```suggestion
  fmt.Println("Workflow ID %v", exec.Execution.WorkflowId)
```

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 12]
```diff
@@ -9,4 +9,26 @@ tags:
   - client
 ---
 
-Use [`Client.ListWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow).
+The [`ListWorkflow()`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow) function retrieves a list of [Workflow Executions](/concepts/what-is-a-workflow-execution) that match the [Search Attributes](/concepts/what-is-a-search-attribute) of a given [List Filter](/concepts/what-is-a-list-filter).
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:12]:
```suggestion
The [ListWorkflow()](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ListWorkflow) function retrieves a list of [Workflow Executions](/concepts/what-is-a-workflow-execution) that match the [Search Attributes](/concepts/what-is-a-search-attribute) of a given [List Filter](/concepts/what-is-a-list-filter).
```

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 13]
```diff
@@ -9,4 +9,26 @@ tags:
...
+The metadata returned from the [Visibility store](/concepts/what-is-visibility) can be used to get a Workflow Execution's history and details from the [Persistence store](/concepts/what-is-a-temporal-cluster#persistence).
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:13]: What do you think about the following changes to link text to more closely match the text to the target?
```suggestion
The metadata returned from the [Visibility](/concepts/what-is-visibility) store can be used to get a Workflow Execution's history and details from the [Persistence](/concepts/what-is-a-temporal-cluster#persistence) store.
```

File: docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md [hunk 14]
```diff
@@ -9,4 +9,26 @@ tags:
...
+
+Use a List Filter to define a `request` to pass into `ListWorkflow()`.
+
+```go
+request := &workflowservice.ListWorkflowExecutionsRequest{ Query: "CloseTime = missing" }
+```
+
+This `request` value returns only open Workflows.
+For more List Filter examples, see the [examples provided for List Filters in Temporal's Visibility guide.](/concepts/what-is-a-list-filter#list-filter-examples)
```
Comment (djmagee) [inline docs-src/go/how-to-list-workflow-executions-using-the-client-in-go.md:22]:
```suggestion
For more List Filter examples, see the [examples provided for List Filters in the Temporal Visibility guide.](/concepts/what-is-a-list-filter#list-filter-examples)
```

**Review by cretz — APPROVED** [review]

**Pull request merged**